packages/devtools_app/lib/src/standalone_ui/api/vs_code_api.dart

```diff
@@ -9,12 +9,46 @@
 /// [VsCodeCapabilities] to advertise which capabilities are available and
 /// handle any changes in behaviour.
 abstract interface class VsCodeApi {
+  /// The capabilities of the instance of VS Code / Dart VS Code extension that
+  /// we are connected to.
+  ///
+  /// All API calls should be guarded by checks of capabilities because the API
+  /// may change over time.
   VsCodeCapabilities get capabilities;
+
+  /// Informs the VS Code extension we are initialized, allowing it to send
+  /// initial events to all streams with an initial set of data.
   Future<void> initialize();
+
+  /// A stream of events for whenever the set of devices (or selected device)
+  /// change in VS Code.
+  ///
+  /// An event with initial devices is sent after [initialize] is called.
   Stream<VsCodeDevicesEvent> get devicesChanged;
+
+  /// A stream of events for whenever the set of debug sessions change or are
+  /// updated in VS Code.
+  ///
+  /// An event with initial sessions is sent after [initialize] is called.
   Stream<VsCodeDebugSessionsEvent> get debugSessionsChanged;
+
+  /// Executes a VS Code command.
+  ///
+  /// Commands can be native VS Code commands or commands registered by the
+  /// Dart/Flutter extensions.
+  ///
+  /// Which commands are available is not part of the API contract so callers
+  /// should take care when calling APIs that might evolve over time.
   Future<Object?> executeCommand(String command, [List<Object?>? arguments]);
+
+  /// Changes the current Flutter device.
+  ///
+  /// The selected device is the same one shown in the status bar in VS Code.
+  /// Calling this API will update the device for the whole VS Code extension.
   Future<bool> selectDevice(String id);
+
+  /// Opens a specific DevTools [page] for the debug session with ID
```

Comment:
Does this open the DevTools page in the embedded vs code window or in the browser? could be good to specify here.

Reply:
It could be either depending on user settings. I've updated the comment to note this.